### PR TITLE
chore: replace lerna by yarn workspace and use talend-scripts

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this Pull Request is trying to solve?**

a lot of security alerts are raised because of lerna 

**What is the chosen solution to this problem?**
 
use yarn workspaces and talend-script

**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-client/package.json [hunk 1]
```diff
@@ -19,41 +17,21 @@
...
-    "jest": "20.0.3",
-    "jest-cli": "20.0.3",
-    "jsdoc": "^3.4.1",
-    "rimraf": "^2.6.1"
+    "@talend/context": "^5.0.2",
+    "@talend/scripts-core": "^11.1.0",
+    "@talend/scripts-preset-react-lib": "^9.9.2",
+    "enzyme": "^3.11.0",
+    "react": "^16.14.0",
+    "react-dom": "^16.14.0",
```
Comment (gjou-tlnd) [inline daikon-tql/daikon-tql-client/package.json:26]: AFAIR we don't have react components in this lib

**Review by gjou-tlnd — CHANGES_REQUESTED** [review]

File: daikon-tql/daikon-tql-client/package.json [hunk 2]
```diff
@@ -19,41 +17,21 @@
...
-    "eslint": "^4.18.2",
-    "eslint-config-airbnb": "^11.0.0",
-    "eslint-plugin-import": "^1.14.0",
-    "eslint-plugin-jsx-a11y": "^2.2.1",
-    "eslint-plugin-react": "^6.2.0",
-    "jest": "20.0.3",
-    "jest-cli": "20.0.3",
-    "jsdoc": "^3.4.1",
-    "rimraf": "^2.6.1"
+    "@talend/context": "^5.0.2",
```
Comment (gjou-tlnd) [inline daikon-tql/daikon-tql-client/package.json:21]: is this used?

File: daikon-tql/daikon-tql-client/package.json (hunk 1 above)
Comment (ncomont) [reply daikon-tql/daikon-tql-client/package.json:26]: image
seems required by talend-scripts when executing tests

File: daikon-tql/daikon-tql-client/package.json (hunk 2 above)
Comment (ncomont) [reply daikon-tql/daikon-tql-client/package.json]: oops nop, fixed thanks

File: daikon-tql/daikon-tql-client/package.json (hunk 1 above)
Comment (gjou-tlnd) [reply daikon-tql/daikon-tql-client/package.json:25]: I guess we need a preset for non react-lib (cc @jmfrancois)

Comment (jmfrancois) [reply daikon-tql/daikon-tql-client/package.json:25]: test-setup require react https://github.com/Talend/ui-scripts/blob/master/packages/config-jest/test-setup.js
not react components.
It is a peerDependencies: https://github.com/Talend/ui-scripts/blob/master/packages/config-jest/package.json#L21

**Review by gjou-tlnd — APPROVED** [review]

File: daikon-tql/daikon-tql-client/package.json (hunk 1 above)
Comment (ncomont) [reply daikon-tql/daikon-tql-client/package.json:25]: thanks,
react-dom is also required (by enzyme)
image

Comment (ncomont) [reply daikon-tql/daikon-tql-client/package.json:25]: thanks,
react-dom is also required (by enzyme)
image

File: daikon-tql/README.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ $> mvn install -Pjavascript
 To publish on npm repository the TQL client and generated grammar, you must be authorized.
 
 ```bash
-$> yarn  publish
+$> yarn npm publish
```
Comment (jmfrancois) [inline daikon-tql/README.md:34]: you must call `yarn pre-release` before as I have seen in the package.json
The script publish do not exists in the root package.json which is private so the effect here will be to try to publish it and just fails.

File: daikon-tql/package.json [hunk 4]
```diff
@@ -3,12 +3,20 @@
...
+    "pre-release": "yarn workspaces run pre-release",
+    "lint": "cross-env TZ=Europe/Paris yarn workspaces run lint"
   },
+  "workspaces": [
+    "daikon-tql-client",
+    "daikon-tql-core/target/generated-modules/tql"
+  ],
   "devDependencies": {
-    "lerna": "^2.9.1"
+    "@changesets/cli": "^2.17.0",
```
Comment (jmfrancois) [inline daikon-tql/package.json:15]: do you plan to setup changset ?

**Review by jmfrancois — APPROVED** [review]
not different from lerna but have to note that you have to commit and push the results of changeset version which remove all changeset files, update the changelog and package.json ;)

File: daikon-tql/package.json [hunk 5]
```diff
@@ -3,12 +3,23 @@
   "description": "Private parent Daikon TQL",
   "license": "Apache-2.0",
   "scripts": {
-    "postinstall": "lerna clean --yes && lerna exec -- yarn --pure-lockfile --no-progress --mutex network",
-    "test": "lerna exec --scope=@talend/daikon-tql-client -- yarn test",
-    "publish": "lerna publish --skip-git --force-publish=*"
+    "postinstall": "yarn workspaces run build",
+    "pre-release": "yarn workspaces run pre-release",
```
Comment (jmfrancois) [inline daikon-tql/package.json:7]: you can remove the pre-release script here you don t need it

**Pull request merged**